packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -2456,6 +2420,202 @@ contract OPContractsManager is ISemver {
+        ISuperchainConfig _superchainConfig
+    )
+        internal
+        pure
+        returns (IOPContractsManagerV2.FullConfig memory)
+    {
+        // Start building the full config.
+        IOPContractsManagerV2.FullConfig memory cfg;
+
+        // Handle salt mixer.
+        cfg.saltMixer = _input.saltMixer;
+
+        // Handle system roles.
+        cfg.proxyAdminOwner = _input.roles.opChainProxyAdminOwner;
+        cfg.systemConfigOwner = _input.roles.systemConfigOwner;
+        cfg.unsafeBlockSigner = _input.roles.unsafeBlockSigner;
+        cfg.batcher = _input.roles.batcher;
+
+        // Handle L2 system configuration.
+        cfg.basefeeScalar = _input.basefeeScalar;
+        cfg.blobBasefeeScalar = _input.blobBasefeeScalar;
+        cfg.gasLimit = _input.gasLimit;
+        cfg.l2ChainId = _input.l2ChainId;
+        cfg.resourceConfig = Constants.DEFAULT_RESOURCE_CONFIG();
+
+        // Handle dispute game configs.
+        cfg.disputeGameConfigs = new IOPContractsManagerV2.DisputeGameConfig[](3);
+        cfg.disputeGameConfigs[0] = IOPContractsManagerV2.DisputeGameConfig({
+            enabled: false, // NOTE: We currently disable FDG on first deploy.
+            initBond: 0, // NOTE: We currently disable FDG on first deploy.
+            gameType: GameTypes.CANNON,
+            gameArgs: abi.encode(
+                IOPContractsManagerV2.FaultDisputeGameConfig({ absolutePrestate: _input.disputeAbsolutePrestate })
+            )
+        });
+        cfg.disputeGameConfigs[1] = IOPContractsManagerV2.DisputeGameConfig({
+            enabled: true,
+            initBond: 0, // NOTE: PDG gets a zero init bond for legacy deployments.
+            gameType: GameTypes.PERMISSIONED_CANNON,
+            gameArgs: abi.encode(
+                IOPContractsManagerV2.PermissionedDisputeGameConfig({
+                    absolutePrestate: _input.disputeAbsolutePrestate,
+                    proposer: _input.roles.proposer,
+                    challenger: _input.roles.challenger
+                })
+            )
+        });
+        cfg.disputeGameConfigs[2] = IOPContractsManagerV2.DisputeGameConfig({
+            enabled: false, // NOTE: We currently disable CKDG on first deploy.
+            initBond: 0, // NOTE: We currently disable CKDG on first deploy.
+            gameType: GameTypes.CANNON_KONA,
+            gameArgs: abi.encode(
+                IOPContractsManagerV2.FaultDisputeGameConfig({
+                    absolutePrestate: Claim.wrap(bytes32(0)) // NOTE: Incorrect prestate but disabled so doesn't matter.
+                 })
+            )
+        });
+
+        // Handle anchor state configuration.
+        cfg.startingAnchorRoot = abi.decode(_input.startingAnchorRoot, (Proposal));
+        cfg.startingRespectedGameType = GameTypes.PERMISSIONED_CANNON;
+
+        // Handle SuperchainConfig.
+        cfg.superchainConfig = _superchainConfig;
+
+        // Return the full config.
+        return cfg;
+    }
```

Comment:
For my own sake summarizing how we handle fields not present on DeployInput, which are added to the FullConfig:
- resourceConfig: uses DEFAULT_RESOURCE_CONFIG
- startingRespectedGameType: uses `GameTypes.PERMISSIONED_CANNON;`
- disputeGameConfigs: includes only the permissioned game, takes the proposer/challenger from DeployInput

So, this ultimately crafts a `FullConfig` which matches the behavior of `deploy()`.